docs/user-guide/masking.ipynb

```diff
@@ -0,0 +1,201 @@
+   "metadata": {},
+   "source": [
+    "## Creating and manipulating masks\n",
+    "\n",
+    "Masks are simply variables with `dtype=bool`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mask = sc.Variable([Dim.X], values=[False, False, True])\n",
+    "sc.table(mask)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Boolean operators can be used to manipulate such variables:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(~mask)\n",
+    "print(mask ^ mask)\n",
+    "print(mask & ~mask)\n",
+    "print(mask | ~mask)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Masks in DataArray and Dataset\n",
+    "\n",
+    "Data arrays and dataset can store arbitrary masks.\n",
+    "This is accessible using the `masks` keyword-argument and property, which behaves in the same way as `coords`, `labels`, and `attrs`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = sc.DataArray(\n",
+    "    data = sc.Variable(dims=[Dim.Y, Dim.X], values=np.arange(1.0, 7.0).reshape((2, 3))),\n",
+    "    coords={\n",
+    "        Dim.Y: sc.Variable([Dim.Y], values=np.arange(2.0), unit=sc.units.m),\n",
+    "        Dim.X: sc.Variable([Dim.X], values=np.arange(3.0), unit=sc.units.m)},\n",
+    "    masks={\n",
+    "        'x': sc.Variable([Dim.X], values=[False, False, True])}\n",
+    "    )\n",
+    "sc.show(a)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "b = a.copy()\n",
+    "b.masks['x'].values[1] = True\n",
+    "b.masks['y'] = sc.Variable([Dim.Y], values=[False, True])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that setting a mask does *not* affect the data."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Operations with masked objects\n",
+    "\n",
+    "### Element-wise binary operations\n",
+    "\n",
+    "The result of operations between data arrays or dataset with masks contains the masks of both inputs.\n",
+    "If both inputs contain a mask with the same name, the output mask is the combination of the input masks with an  **OR** operation:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a + b"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Reduction operations\n",
+    "\n",
+    "Operations like `sum` and `mean` over a particular dimension cannot preserve masks that depend on this dimension.\n",
+    "If this is the case, the mask is applied during the operation and is not present in the output:"
```

Comment:
I think the description and examples here are good and clarify an existing assumption I had with the declaration of the masks - that they might broadcast over the other dimensions.